crates/engine/tree/src/tree/payload_processor/multiproof.rs

```diff
@@ -997,147 +1016,164 @@ impl MultiProofTask {
+
+                            self.metrics
+                                .proof_calculation_duration_histogram
+                                .record(proof_result.elapsed);
+
+                            self.multiproof_manager.on_calculation_complete();
+
+                            // Convert ProofResultMessage to SparseTrieUpdate
+                            match proof_result.result {
+                                Ok((multiproof, _stats)) => {
+                                    debug!(
+                                        target: "engine::tree::payload_processor::multiproof",
+                                        sequence = proof_result.sequence_number,
+                                        total_proofs = proofs_processed,
+                                        "Processing calculated proof from worker"
+                                    );
+
+                                    let update = SparseTrieUpdate {
+                                        state: proof_result.state,
+                                        multiproof,
+                                    };
+
+                                    if let Some(combined_update) =
+                                        self.on_proof(proof_result.sequence_number, update)
+                                    {
+                                        let _ = self.to_sparse_trie.send(combined_update);
+                                    }
+                                }
+                                Err(error) => {
+                                    error!(target: "engine::tree::payload_processor::multiproof", ?error, "proof calculation error from worker");
+                                    return
+                                }
+                            }
+
+                            if self.is_done(
+                                proofs_processed,
+                                state_update_proofs_requested,
+                                prefetch_proofs_requested,
+                                updates_finished,
+                            ) {
+                                debug!(
+                                    target: "engine::tree::payload_processor::multiproof",
+                                    "State updates finished and all proofs processed, ending calculation"
+                                );
+                                break
+                            }
+                        }
+                        Err(_) => {
+                            // SAFETY: This is unreachable because `self.multiproof_manager` owns
+                            // `proof_result_tx` (the sender), which keeps the channel open for
+                            // the entire lifetime of this task. The channel cannot close while
+                            // `self` exists.
+                            unreachable!("proof result channel closed while multiproof manager still exists")
```

Comment:
this checks out because select does return `RecvError` and not TryRecvError so this is only reachable if the channel is disconnected.

tho maybe we should just return the an error here